tasks/README.md

```diff
@@ -104,8 +130,45 @@ If a parameter type accepts `null` the task runner will accept either a `null` v
 
 ### Metaparameters
 
-In addition to the tasks parameters, the task runner may inject metaparameters prefixed by '_'. These include `_noop` and `_task`.
+In addition to the tasks parameters, the task runner may inject metaparameters prefixed by `_`.
+
+**_noop**: Used to implement logic in tasks to support cases where task should not perform certain actions.
+
+**_task**: Allow multiple task implementations to access the same executable file. The `_task` metaparameter provides the executable the task name to allow task specific logic to be implemented. 
+
+**_installdir**: Tasks with `files` specified in the metatdata will be passed the `_installdir` metaparam to provide the file path to the expected executables. 
 
+#### Metaparameter Examples
+
+When the task runner runs a task with files it copies those files into the same temporary directory as the task. The task can then load the code based on the path. The directory structure of the specified file resources will be preserved such that paths specified with the `files` metatdata option will be available to tasks prefixed with `_installdir` and relative paths will work as expected. The examples below illustrate this behaviour. 
+
+##### Using `_installdir`
+###### Metadata
+```json
+{ "files": "mymodule/files/script1.py" }
+```
+###### Task
+```python
+require json
+params = JSON.parse(STDIN.read)
+exec("#{params['_installdir']}/mymodule/files/script1.py", "--flag", params["flag1"], "-f", params["flag2"], params[arg1])
+```
+##### Using relative path
+###### Metadata
+```json
+{ "files": "task_helpers/files/ruby.rb" }
+```
+###### Task
+```ruby
+require_relative "../../task_helpers/files/ruby.rb"
```

Comment:
I'm not sure I'd include a relative example, I think we want more freedom around how to structure the directory where we copy over files.